beacon-chain/core/blocks/block_operations.go

```diff
@@ -721,70 +678,15 @@ func ProcessValidatorDeposits(
 	}
 
 	valIndexMap := stateutils.ValidatorIndexMap(beaconState)
```

Comment:
After speaking with Nishant we decided to move generating the mapping to outside the loop, although I'm worried this would cause problems.

What if an actor sent 2 32 ETH deposits from the same public key during the same block? Wouldn't it process the first, add it to the registry, and since the mapping isn't updated on each deposit it'll add the second deposit to the registry as well?